Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -162,8 +162,8 @@ static bool shouldAllowNonInteractiveCursorForElement(const Element& element)
         return true;
 #endif
 
-    if (is<HTMLTextFormControlElement>(element))
-        return !element.focused();
+    if (auto* textElement = dynamicDowncast<HTMLTextFormControlElement>(element))
+        return !textElement->focused() || !textElement->lastChangeWasUserEdit() || textElement->value().isEmpty();
```

Comment:
You should be able to test the new logic here with `UIHelper.activateAndWaitForInputSession*()` and `UIHelper.typeCharacter()`. We still want to cover cases where we hide the region!

Reply:
Added 2 new tests:  `textarea-focused-edited.html` and `text-input-focused-edited.html`.